src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1674,15 +1674,16 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                     BoundNode methodBody = bodyBinder.BindMethodBody(syntaxNode, diagnostics);
                     BoundNode methodBodyForSemanticModel = methodBody;
                     NullableWalker.SnapshotManager snapshotManager = null;
+                    ImmutableDictionary<Symbol, Symbol> remappedSymbols = null;
                     if (bodyBinder.Compilation.NullableSemanticAnalysisEnabled)
                     {
                         // Currently, we're passing an empty DiagnosticBag here because the flow analysis pass later will
                         // also run the nullable walker, and issue duplicate warnings. We should try to only run the pass
                         // once.
                         // https://github.com/dotnet/roslyn/issues/35041
-                        methodBodyForSemanticModel = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: true, out snapshotManager);
+                        methodBodyForSemanticModel = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: true, out snapshotManager, out remappedSymbols);
```

Comment:
>methodBodyForSemanticModel = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: true, out snapshotManager, out remappedSymbols); [](start = 24, length = 204)

Do we have an issue tracking making this work lazy?

Reply:
[https://github.com/dotnet/roslyn/issues/35041](https://github.com/dotnet/roslyn/issues/35041)

---
In reply to: [329704812](https://github.com/dotnet/roslyn/pull/38769#discussion_r329704812) [](ancestors = 329704812)